### PR TITLE
DAOS-10015 build: add diffutils to Dockerfile.checkpatch

### PR DESCRIPTION
**Pull request by liuxuezhao**
Signed-off-by: Xuezhao Liu <xuezhao.liu@intel.com>

### CODE REVIEW

File: utils/docker/Dockerfile.checkpatch [hunk 1]
```diff
@@ -15,8 +15,8 @@ ARG CB0
 # Clean up any repos afterwards to save space.
 RUN dnf -y install dnf-plugins-core &&				\
     dnf config-manager --save --setopt=install_weak_deps=False && \
-    dnf -y upgrade &&						\
-    dnf -y install codespell file findutils git golang-bin	\
+    dnf -y upgrade &&							\
+    dnf -y install codespell file findutils git golang-bin diffutils	\
```
Comment (ashleypittman) [inline utils/docker/Dockerfile.checkpatch:19]: These are ordered alphabetically currently but it's not a big issue.

**Review by ashleypittman — COMMENTED** [review]
This is fine, as long as it can be shown to solve a genuine problem, can you provide a link to a build where this is required?

It's likely that this command may only be needed based on what's in the build, for example a PR which is "clean" and has no checkpatch warnings (line this one) may not need the patch, but one with syntax/spelling/checkpatch warnings may need it.

**Comment by liuxuezhao** [comment]
> This is fine, as long as it can be shown to solve a genuine problem, can you provide a link to a build where this is required?
> 
> It's likely that this command may only be needed based on what's in the build, for example a PR which is "clean" and has no checkpatch warnings (line this one) may not need the patch, but one with syntax/spelling/checkpatch warnings may need it.

https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/PR-8338/runs/4/log/?start=0
is this what you want?

**Comment by ashleypittman** [comment]
> > This is fine, as long as it can be shown to solve a genuine problem, can you provide a link to a build where this is required?
> > It's likely that this command may only be needed based on what's in the build, for example a PR which is "clean" and has no checkpatch warnings (line this one) may not need the patch, but one with syntax/spelling/checkpatch warnings may need it.
> 
> https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/PR-8338/runs/4/log/?start=0 is this what you want?

I'm ill today so not working or on the VPN but I'll have a look tomorrow.

**Review by ashleypittman — APPROVED** [review]
This won't affect the test failures in https://github.com/daos-stack/daos/pull/8338 however it does resolve some problems with checkpatch so we do want this PR.

It's likely this changed in the docker server upgrade a couple of weeks ago, and PRs will not have been getting checkpatch.pl feedback in that time.

**Comment by liuxuezhao** [comment]
> This won't affect the test failures in #8338 however it does resolve some problems with checkpatch so we do want this PR.
> 
> It's likely this changed in the docker server upgrade a couple of weeks ago, and PRs will not have been getting checkpatch.pl feedback in that time.

hmm, PR8338 new build (with this patch together), still with NLT failure reported (not same with before). I failed to find what's the problem, could you please take a look if convenient? thanks.
https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/PR-8338/runs/5/log/?start=0

**Review by jolivier23 — APPROVED** [review]

**Comment by liuxuezhao** [comment]
CI test failure due to DAOS-9875.

**Comment by ashleypittman** [comment]
We probably still wanted this didn't we?

**Pull request closed without merging**